azext_edge/edge/providers/check/base/pod.py

```diff
@@ -27,229 +33,191 @@ def decorate_pod_phase(phase: str) -> Tuple[str, str]:
+        table = Table.grid(padding=(0, 0, 0, 2))
+
+    add_footer = False
+    pod_statuses = []
 
-        # When pod in obnormal state, sometimes the conditions are not available
-        if pod_conditions:
-            for condition in pod_conditions:
-                type = condition["type"]
-                condition_type = POD_CONDITION_TEXT_MAP.get(type)
-
-                if condition_type:
-                    condition_status = condition.get("status") == "True"
-                    conditions_readiness = conditions_readiness and condition_status
-                    pod_condition_deco, status = _decorate_pod_condition(condition=condition_status)
-                    pod_eval_status = status if status != CheckTaskStatus.success.value else pod_eval_status
-                else:
-                    condition_type = type
-                    condition_status = condition.get("status")
-
-                formatted_reason = ""
-                condition_reason = condition.get("reason", "")
-
-                if condition_reason:
-                    formatted_reason = f"[red]Reason: {condition_reason}[/red]"
-
-                known_condition_values = [value.replace(" ", "").lower() for value in POD_CONDITION_TEXT_MAP.values()]
-                if condition_type.replace(" ", "").lower() in known_condition_values:
-                    conditions_display_list.append((f"{condition_type}: {pod_condition_deco}", formatted_reason))
-                else:
-                    unknown_conditions_display_list.append((f"{condition_type}: {condition_status}", formatted_reason))
-
-                pod_eval_value[f"status.conditions.{type.lower()}"] = condition_status
-
-        if not conditions_readiness:
-            pod_eval_status = CheckTaskStatus.error.value
-        else:
-            # add warning if there are unknown conditions when known conditions are all in good state
-            if unknown_conditions_display_list and pod_eval_status != CheckTaskStatus.error.value:
-                pod_eval_status = CheckTaskStatus.warning.value
-
-        _add_pod_health_display(
+    for pod in pods:
+        pod_status_result: PodStatusResult = _process_pod_status(
             check_manager=check_manager,
             target=target,
+            pod=pod,
             namespace=namespace,
-            pod_name=pod_name,
-            pod_phase_deco=pod_phase_deco,
-            pod_conditions=pod_conditions,
-            pod_eval_status=pod_eval_status,
-            conditions_readiness=conditions_readiness,
-            conditions_display_list=conditions_display_list,
-            unknown_conditions_display_list=unknown_conditions_display_list,
-            display_padding=display_padding,
             detail_level=detail_level,
         )
+        pod_statuses.append(pod_status_result.eval_status)
+        table.add_row(*pod_status_result.display_strings)
+
+    add_footer = not all([status == CheckTaskStatus.success.value for status in pod_statuses])
```

Comment:
This should fix the linter issue (please double-check) - it's asking us to use a generator instead of creating a new comprehension object to pass to `all`
```suggestion
    add_footer = not all(status == CheckTaskStatus.success.value for status in pod_statuses)
```

Reply:
oh I see, thanks a lot!